setup.py

```diff
@@ -974,8 +974,53 @@ def print_box(msg):
         print('|{}{}|'.format(l, ' ' * (size - len(l))))
     print('-' * (size + 2))
 
+def pre_build_checks():
+    if sys.version_info >= (3, 11):
```

Comment:
Plz dismantle pyramid of doom :)
```suggestion
    if sys.version_info < (3, 11):
        return
```

Reply:
That was to allow more to be added below in this function. But I guess I'll make it more specific and reduce nesting!